cornflow-server/cornflow/endpoints/login.py

```diff
@@ -55,10 +51,24 @@ def log_in(self, **kwargs):
         if auth_type == AUTH_DB:
             user = self.auth_db_authenticate(**kwargs)
             response.update({"change_password": check_last_password_change(user)})
+            current_app.logger.info(f"User {user.email} logged in successfully using database authentication")
         elif auth_type == AUTH_LDAP:
             user = self.auth_ldap_authenticate(**kwargs)
+            current_app.logger.info(f"User {user.email} logged in successfully using LDAP authentication")
         elif auth_type == AUTH_OID:
-            user = self.auth_oid_authenticate(**kwargs)
+            if (kwargs.get('username') and kwargs.get('password')):
+                if not current_app.config.get("SERVICE_USER_ALLOW_PASSWORD_LOGIN", 0):
+                    raise InvalidUsage("Must provide a token in Authorization header. Cannot log in with username and password", 400)
+                user = self.auth_oid_authenticate(username=kwargs['username'], password=kwargs['password'])
+                current_app.logger.info(f"Service user {user.email} logged in successfully using password")
+                token = self.auth_class.generate_token(user.id)
+            else:
+                token = self.auth_class().get_token_from_header(request.headers)
+                user = self.auth_oid_authenticate(token=token)
+                current_app.logger.info(f"User {user.email} logged in successfully using OpenID authentication")
+            
+            response.update({"token": token, "id": user.id})
```

Comment:
I would make the token optional on the response. I would only add the token on the body for the service user, for the rest of users i would just give them back their `user.id` as it can be useful to know on the client side